nan.h

```diff
@@ -1527,26 +1755,28 @@ class NanObjectWrap : public node::ObjectWrap {
 template <typename T, typename P>
 class NanWeakCallbackData {
  public:  // constructors
+  inline ~NanWeakCallbackData();
+
+ private:  // constructors
+  NAN_DISALLOW_ASSIGN_COPY_MOVE(NanWeakCallbackData)
+
+ public:
   typedef void (*Callback)(
       NanWeakCallbackData & data  // NOLINT(runtime/references)
       );
-  NanWeakCallbackData(v8::Handle<T> handle, P* param, Callback cb)
-    : parameter(param), callback(cb) {
-    NanAssignPersistent(persistent, handle);
-    Revive();
+  NanWeakCallbackData(v8::Persistent<T> *handle, P* param, Callback cb)
+    : persistent(), parameter(param), callback(cb) {
+    std::memcpy(&persistent, handle, sizeof(persistent));
```

Comment:
But it is a trivial class. All it contains is a T *value_;

On May 18, 2015 10:23:38 PM EEST, Ben Noordhuis <notifications@github.com> wrote:
>>    typedef void (*Callback)(
>>        NanWeakCallbackData & data  // NOLINT(runtime/references)
>>        );
>> -  NanWeakCallbackData(v8::Handle<T> handle, P* param, Callback cb)
>> -    : parameter(param), callback(cb) {
>> -    NanAssignPersistent(persistent, handle);
>> -    Revive();
>> +  NanWeakCallbackData(v8::Persistent<T> *handle, P* param, Callback
>cb)
>> +    : persistent(), parameter(param), callback(cb) {
>> +    std::memcpy(&persistent, handle, sizeof(persistent));
>
>Excuses le mot but memcpy'ing the persistent handle seems like a
>terrible idea, never mind that memcpy'ing non-trivial classes is nearly
>always a bad idea in general.  It's actively subverting how persistent
>handles are supposed to work and it could break on a whim.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/nodejs/nan/pull/348/files#r30539047